content/articles/creating-serverless-functions-faas-with-aws-lambda-and-golang/index.md

```diff
@@ -0,0 +1,292 @@
+
+}
+
+ 
+
+func main() {
+
+ lambda.Start(HandleInfoEvent)
+
+}
+
+```
+
+You can also run `git clone https://github.com/Bamimore-Tomi/faas-golang.git` and navigate to the “server” folder to see this source code. In the program above, we created the `HandleInfoEvent` function. This is the function that runs in the cloud. Use the following steps to compile and zip the program.
+
+#### Macos and Linux Installation Instructions
+
+1. Compile the executable.
+
+```bash
+
+GOOS=linux go build main.go
+
+```
+
+2. Create the zip file with
+
+```bash
+
+zip user-profile-function.zip main
+
+```
+
+#### Windows Installation Instructions
+
+1. Download the `build-lambda-zip` tool from GitHub
+
+```bash
+
+go.exe get -u github.com/aws/aws-lambda-go/cmd/build-lambda-zip
+
+```
+
+2. You can now use the tool downloaded earlier to create the zip file.
+
+```bash
+
+Set GOOS=linux
+
+Go build -o main main.go
+
+%USERPROFILE%\Go\bin\build-lambda-zip.exe -output main.zip main
+
+```
+
+You can now upload the zip file generated. You can test the function you uploaded right on AWS to see if it works well. First, we need to change the handler's name from the default `hello` to `main`. The Lambda function handler is the method in your function code that processes events.
+
+![aws-lambda-edit-hanler](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/aws-lambda-edit-handler.png)
+
+Click on Edit:
+
+![aws-lambda-runtime-settings](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/aws-lambda-runtime-settings.png)Change the handler field from hello to main and save.
+
+### Testing the Function on AWS Lambda
+
+1. On the main dashboard, click on Test![aws-lambda-dashboard-test](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/aws-lambda-test.png)
+2. Input this sample data in the json editor on the page and click on Test.
+
+```json
+
+{
+
+"firstname":"Oluwatomisin",
+
+"lastname":"Bamimore",
+
+"age":19
+
+}
+
+```
+
+![aws-lambda-test-prompt](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/aws-lambda-test-prompt.png)
+
+The function we uploaded is executed. This is the result.![aws-lambda-test-result](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/aws-lambda-test-result.png)
+
+### Calling the Lambda Function From Golang
+
+We can also execute the function uploaded to AWS lambda locally. We have to install the AWS Go SDK.
+
+```bash
+
+go get github.com/aws/aws-sdk-go
+
+```
+
+We can now use this code segment to execute the function in the cloud and see the results locally.
+
+```go
+
+package main
+
+ 
+
+import (
+
+ "encoding/json"
+
+ "fmt"
+
+ "os"
+
+ 
+
+ "github.com/aws/aws-sdk-go/aws"
+
+ "github.com/aws/aws-sdk-go/service/lambda"
+
+ "github.com/joho/godotenv"
+
+ 
+
+ "github.com/aws/aws-sdk-go/aws/credentials"
+
```

Comment:
Format this code. Add comments and explain them

Reply:
Alright, thank you.

Reply:
@Bamimore-Tomi 

I don't think this is resolved yet.

By formatting the code, I meant to remove the extra new lines, and follow right indentation.

Reply:
Do this change for all the code snippets